gossip/src/cluster_info.rs

```diff
@@ -1014,15 +1015,32 @@ impl ClusterInfo {
                 }
             }
         };
+        // Returns true if the vote has the same slot as the transaction
+        let is_duplicate_slot = |vote: &Vote| -> bool {
+            if let Some(slot) = vote.slot() {
+                return slot == vote_slot;
+            }
+            false
+        };
```

Comment:
This only makes the code more verbose and less readable.
Instead you can just do
```rust
CrdsData::Vote(_, vote) if vote.slot() == Some(vote_slot) => ...
```
below in the match statement.